From ed7a58bcdcf62fbd35baed2504e9f84e91a66f22 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Alex=20R=C3=B8nne=20Petersen?= <alex@alexrp.com>
Date: Thu, 18 Sep 2025 00:39:15 +0200
Subject: [PATCH 077/133] test: disable some vector ctz/clz behavior tests on
 LoongArch with LSX
X-Developer-Signature: v=1; a=openpgp-sha256; l=1021; i=xtex@astrafall.org;
 h=from:subject; bh=HWxsDo/1+FVpR9a0fx2utYd8dw05H0nh1lz/2rLQYVk=;
 b=owGbwMvMwCW2U4Ij7wZL9ETG02pJDJn78thcvp9v41kTeXJ7o8iJjArnpM/fxAQTdWded3w/0
 S76bcP+joksDGJcDJZiiixFhg3erDrp/KLLymVh5rAygQyRFmlgYGBgYGHgy03MKzXSMdIz1TbU
 MzTUMdIxYuDiFICpbi1lZDgTeSd7R6nI9KzfAUy6v8811F96cdPZ/Penayq3NUwm8cgw/E9ccsD
 gyYq0DxsefOcKavqg7L/bsjg7iKfwxg2lKqat2jwA
X-Developer-Key: i=xtex@astrafall.org; a=openpgp;
 fpr=7231804B052C670F15A6771DB918086ED8045B91

https://github.com/llvm/llvm-project/issues/159529
(cherry picked from commit beb25b0430d96e2e87f6af189062bdd3a02b6e45)
---
 test/behavior/math.zig | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/test/behavior/math.zig b/test/behavior/math.zig
index 679c94a78e39..914c47ba69c2 100644
--- a/test/behavior/math.zig
+++ b/test/behavior/math.zig
@@ -111,6 +111,8 @@ test "@clz vectors" {
 }
 
 fn testClzVectors() !void {
+    if (comptime builtin.cpu.has(.loongarch, .lsx)) return error.SkipZigTest; // https://github.com/llvm/llvm-project/issues/159529
+
     const Vu4 = @Vector(64, u4);
     const Vu8 = @Vector(64, u8);
     const Vu128 = @Vector(64, u128);
@@ -197,6 +199,8 @@ test "@ctz vectors" {
 }
 
 fn testCtzVectors() !void {
+    if (comptime builtin.cpu.has(.loongarch, .lsx)) return error.SkipZigTest; // https://github.com/llvm/llvm-project/issues/159529
+
     const Vu4 = @Vector(64, u4);
     const Vu8 = @Vector(64, u8);
     @setEvalBranchQuota(10_000);
-- 
2.52.0

